src/components/dashboard.js

```diff
@@ -121,105 +111,100 @@ export default function Dashboard(props) {
   const handleDrawerClose = () => {
     setOpen(false);
   };
-  const fixedHeightPaper = clsx(classes.paper, classes.fixedHeight);
 
   return (
     <div className={classes.root}>
       <CssBaseline />
-      <UpperBar renderDrawer={props.renderDrawer} classes={classes} open={open} handleDrawerOpen={handleDrawerOpen} />
-      <LeftDrawer renderDrawer={props.renderDrawer} classes={classes} open={open} handleDrawerClose={handleDrawerClose} />
-      <MainContent classes={classes} fixedHeightPaper={fixedHeightPaper} />
+      <UpperBar
+        renderDrawer={props.renderDrawer}
+        classes={classes}
+        open={open}
+        handleDrawerOpen={handleDrawerOpen}
+      />
+      <LeftDrawer
+        renderDrawer={props.renderDrawer}
+        location={props.location}
+        classes={classes}
+        open={open}
+        handleDrawerClose={handleDrawerClose}
+      />
     </div>
   );
 }
 
 function UpperBar({ renderDrawer, classes, open, handleDrawerOpen }) {
-  return (
-    <AppBar position="absolute" className={clsx(classes.appBar, open && classes.appBarShift)}>
-      <Toolbar className={classes.toolbar}>
-        {renderDrawer &&
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const handleMenu = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+  const profileOpen = Boolean(anchorEl);
+  const renderElement = () => {
+    if (renderDrawer) {
```

Comment:
Suggestion (not sure a good idea): Separate different logic with extra blank line, e.g.
```javascript
const handleMenu = (event) => {
  ...
}

const handleClose = () => {
  ...
}
```

Probably look neater this way (?)

Reply:
We will just use Prettier for code consistency, maybe before we deploy our final product.